Content.Server/CartridgeLoader/CartridgeLoaderSystem.cs

```diff
@@ -79,17 +84,37 @@ public List<EntityUid> GetAvailablePrograms(EntityUid uid, CartridgeLoaderCompon
         if (!Resolve(uid, ref loader))
             return new List<EntityUid>();
 
-        //Don't count a cartridge that has already been installed as available to avoid confusion
-        if (loader.CartridgeSlot.HasItem && IsInstalled(Prototype(loader.CartridgeSlot.Item!.Value)?.ID, loader))
+        var unsecureAllowed = _configManager.GetCVar(CCVars.CrewManifestUnsecure);
+        var hasCartridge = loader.CartridgeSlot.HasItem;
+
+        // Don't count a cartridge that has already been installed as available to avoid confusion
+        if (unsecureAllowed && hasCartridge && IsInstalled(Prototype(loader.CartridgeSlot.Item!.Value)?.ID, loader))
             return loader.InstalledPrograms;
 
-        var available = new List<EntityUid>();
-        available.AddRange(loader.InstalledPrograms);
+        // Prepare capacity for InstalledPrograms + CartridgeSlot
+        var programs = new List<EntityUid>(loader.InstalledPrograms.Count + 1);
+
+        if (unsecureAllowed)
+        {
+            programs.AddRange(loader.InstalledPrograms);
+            if (loader.CartridgeSlot.HasItem)
+                programs.Add(loader.CartridgeSlot.Item!.Value);
+
+            return programs;
+        }
+
+        // Unsecure manifest viewers are not allowed
+        // Any CrewManifestViewer should be excluded from the final result
+        foreach (var installedProgram in loader.InstalledPrograms)
+        {
+            if (!TryComp(installedProgram, out CrewManifestViewerComponent? _))
+                programs.Add(installedProgram);
+        }
 
-        if (loader.CartridgeSlot.HasItem)
-            available.Add(loader.CartridgeSlot.Item!.Value);
+        if (loader.CartridgeSlot.HasItem && !TryComp(loader.CartridgeSlot.Item!.Value, out CrewManifestViewerComponent? _))
```

Comment:
the generic cartridge loader system shouldnt have special-case stuff for specific cartridges like this

Reply:
What would you suggest?

Reply:
What would you propose?

Reply:
Reconsidered the approach. Now cartridge loader system doesn't know anything about special cases. Thank you.